protos/flyteidl/core/tasks.proto

```diff
@@ -61,6 +61,22 @@ message RuntimeMetadata {
     string flavor = 3;
 }
 
+// Metadata associated with the GPU accelerator to allocate to a task
+message GPUAccelerator {
+    string device = 1;
+    oneof partition_size_value {
+        bool unpartitioned = 2;
+        string partition_size = 3;
+    }
+}
+
+// Additional metadata associated with resources to allocate to a task
+message ResourceMetadata {
+    oneof accelerator_value {
```

Comment:
Why oneof? if it's not set, it'll just be nil, do you want to allow nullifying a field? if so maybe that should go into the override message?

Reply:
The thought was to support other accelerator types like TPUs in the future. One option would have been to just add all possible accelerator types as separate fields without the guarantee that oneof provides - that only one type is set at any given point in time.

I actually did not consider nullifying an accelerator specified in the task metadata, in the override. That does sound nice. If we were to do that, would it be a bad idea to add a null_accelerator to the oneof?

Reply:
Since oneofs do not change wire format, I would keep it out and kick the can down the road when we add a different accelerator to decide then.

There isn't a great way to nullify in Protos unfortunately...

Reply:
https://github.com/flyteorg/flyteidl/pull/442